src/components/common/InputWithButton.tsx

```diff
@@ -0,0 +1,43 @@
+import { INPUT_WITH_BUTTON_STYLE } from '@/constants/styles';
+
+type InputWithButtonProps = {
+  placeholder?: string;
+  inputText?: string;
+  buttonText?: string;
+  onChange: (inputText: string) => void;
+  onSubmit: () => void;
+};
+
+const InputWithButton = (props: InputWithButtonProps) => {
+  const {
+    placeholder = '검색어를 입력해 주세요.',
+    inputText,
+    buttonText = '확인',
+    onChange,
+    onSubmit,
+  } = props;
+
+  const handleKeyDown = (e: React.KeyboardEvent) => {
+    if (e.key === 'Enter') {
+      onSubmit();
+    }
+  };
+
+  return (
+    <div className={INPUT_WITH_BUTTON_STYLE.container}>
+      <input
+        type="text"
+        value={inputText}
+        placeholder={placeholder}
+        onChange={(e) => onChange(e.target.value)}
```

Comment:
```suggestion
        onChange={onChange}
```
지금 InputWithButton은 common에 위치하고 있습니다. 즉, 공통 컴포넌트로 만들어 두셨다는건데, 그럼 정말 열어두게 설계를 해야한다고 저는 생각합니다.
물론 대부분은 `e.target.value`를 쓰겠지만, 누구는 `e.target.innerText` 값을 다루고 싶을 수도 있습니다.
그러니 어느 특정 페이지에 속하는 컴포넌트가 아니고 이렇게 공통으로 열어두는 컴포넌트라면, 이 컴포넌트를 사용하는 사용자에게 조금 열어두는 방향이 더 좋지 않을까요??

Reply:
리뷰 감사드립니다. 🙏🏻
경태님 말씀대로 event.target의 속성을 고정해두면 재사용성과 확장성이 떨어질 수 있겠네요.